sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/AggregateScanConverter.java

```diff
@@ -240,18 +244,22 @@ private AggregateCall convertAggCall(
     }
 
     List<Integer> argList = new ArrayList<>();
-    for (ResolvedExpr expr :
-        ((ResolvedAggregateFunctionCall) computedColumn.getExpr()).getArgumentList()) {
+    ResolvedAggregateFunctionCall expr = ((ResolvedAggregateFunctionCall) computedColumn.getExpr());
+    List<ZetaSQLResolvedNodeKind.ResolvedNodeKind> resolvedNodeKinds =
+        Arrays.asList(RESOLVED_CAST, RESOLVED_COLUMN_REF, RESOLVED_GET_STRUCT_FIELD);
+    for (int i = 0; i < expr.getArgumentList().size(); i++) {
       // Throw an error if aggregate function's input isn't either a ColumnRef or a cast(ColumnRef).
       // TODO: is there a general way to handle aggregation calls conversion?
-      if (expr.nodeKind() == RESOLVED_CAST
-          || expr.nodeKind() == RESOLVED_COLUMN_REF
-          || expr.nodeKind() == RESOLVED_GET_STRUCT_FIELD) {
+      ZetaSQLResolvedNodeKind.ResolvedNodeKind resolvedNodeKind =
+          expr.getArgumentList().get(i).nodeKind();
+      if (i == 0 && resolvedNodeKinds.contains(resolvedNodeKind)) {
```

Comment:
I added both tests cases and also change the condition to throw an exception in the case of a literal as the first argument. Both tests fall in this case https://ci-beam.apache.org/job/beam_PreCommit_SQL_Commit/4616/.
The tests also failed before changing the condition but with a different exception

Reply:
What was the other exception? Also, do you think it would be possible to support literals as the first argument? I'm not sure why that limitation exists.

Reply:
Both are IndexOutOfBounds when inferring types https://ci-beam.apache.org/job/beam_PreCommit_SQL_Commit/4618/
I think it would be possible to support literals as the first argument, but we should verify every function/case.

Reply:
Alright, for now let's do this:

- Throw an UnsupportedOperationException when literals are passed as the first argument. In other words, keep the current behavior.
- Remove the new array_agg and timestamp test cases.
- We can add support for literals later if we think it's important. I filed a separate JIRA: https://issues.apache.org/jira/browse/BEAM-13648